### PR TITLE
os/bluestore: rework vselector calls.

### PR DESCRIPTION
**Pull request by ifed01**
Plus refactor spillover test cases.

Signed-off-by: Igor Fedotov <igor.fedotov@croit.io>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by aclamk** [comment]
jenkins test make check

File: src/os/bluestore/BlueFS.h [hunk 1]
```diff
@@ -85,10 +85,22 @@ class BlueFSVolumeSelector {
...
+  }
+  void sub_usage(void* hint, const bluefs_fnode_t& fnode) {
+    for (auto& e : fnode.extents) {
+      sub_usage(hint, e);
+    }
+    sub_usage(hint, fnode.size, true);
+  }
+  virtual void add_usage(void* hint, const bluefs_extent_t& extent) = 0;
+  virtual void sub_usage(void* hint, const bluefs_extent_t& extent) = 0;
+  virtual void add_usage(void* hint, uint64_t fsize, bool upd_files = false) = 0;
```
Comment (aclamk) [inline src/os/bluestore/BlueFS.h:102]: I think we need some description to this interface call.

**Review by aclamk — APPROVED** [review]
But I would appreciate description for vselector interface calls.

**Comment by ifed01** [comment]
> ctor interface cal

done

**Comment by NitzanMordhai** [comment]
rados approved

Unrelated failures:
1. https://tracker.ceph.com/issues/63748
2. https://tracker.ceph.com/issues/61774
3. https://tracker.ceph.com/issues/63783
4. https://tracker.ceph.com/issues/59380
5. https://tracker.ceph.com/issues/59142
6. https://tracker.ceph.com/issues/59196


Details:
1. ['7487680', '7487836'] - qa/workunits/post-file.sh: Couldn't create directory
2. ['7487541', '7487610', '7487750', '7487751', '7487683', '7487822'] - centos 9 testing reveals rocksdb "Leak_StillReachable" memory leak in mons
3. ['7487677', '7487816', '7487532'] - mgr: 'ceph rgw realm bootstrap' fails with KeyError: 'realm_name'
4. ['7487804', '7487806', '7487647'] - rados/singleton-nomsgr: test failing from "Health check failed: 1 full osd(s) (OSD_FULL)" and "Health check failed: 1 filesystem is offline (MDS_ALL_DOWN)"
5. ['7487724', '7487568'] - mgr/dashboard: fix e2e for dashboard v3
6. ['7487579', '7487739'] - cephtest bash -c ceph_test_lazy_omap_stats

**Comment by ifed01** [comment]
jenkins test make check

**Pull request merged**